runtime/runtime.c

```diff
@@ -253,9 +253,41 @@ void* allocate_n_bytes_ptr(u32 n) {
 WEAK void populate_globals() {}
 
 // Code that actually runs the wasm code
-IMPORT i32 wasmf_main(i32 a, i32 b);
+IMPORT void wasmf__start(void);
+
```

Comment:
Overall feedback: you've taken apart the abstraction layer here (that strives to make this platform agnostic, and support wasmception and wasi-sdk)

I think it's not too much work to instead preserve this and instead improve the abstraction. YMMV